lib/controllers/RelationalController.ts

```diff
@@ -434,23 +434,34 @@ export abstract class RelationalController extends CRUDController {
    * 
    * @param index
    * @param collection
-   * @param document
+   * @param documentId
     * @param nestedFields : contain field name in the document to get, collection name and index name of documents that are linked
    */
   protected async genericGet<TKDocumentContent extends KDocumentContent> (index: string, collection : string, documentId : string, nestedFields : FieldPath[] = []): Promise<KDocument<TKDocumentContent>> {
     const document = await this.sdk.document.get<TKDocumentContent>(index, collection, documentId);
+    const promises : Promise<void>[] = [];
     for (const nestedField of nestedFields) {
-      if (document._source[nestedField.field]) {
-        document._source[nestedField.field] = await this.getRequestRaw(nestedField.index, nestedField.collection, document._source[nestedField.field]);
-      }
+      promises.push(this.replaceNestedFieldByContent(document, nestedField));
```

Comment:
Instead of declaring a new function, you could use a promise chain. It helps to reduce the cognitive complexity of having to jump in another place of the code 
```
for (const nestedField of nestedFields) {
  if (document._source[nestedField.field]) {
    promises.push(
      this.getRequestRaw(nestedField.index, nestedField.collection, document._source[nestedField.field])
        .then(request => document._source[nestedField.field] = request)
    );
  }
}
```

Reply:
I begin with something like : 
```
  await Promise.all(nestedFields.map(async (nestedField) => {
             document._source[nestedField.field] = await this.getRequestRaw(nestedField.index, nestedField.collection, document._source[nestedField.field]); 
  }));
```

but I wasn't sure it was clearer.